0-bootstrap/modules/groups/main.tf

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module "required_group" {
+  for_each = tomap(var.create_groups_holder.required_groups)
+  source   = "terraform-google-modules/group/google"
+  version  = "~> 0.1"
+
+  id                   = each.value
+  display_name         = each.key
+  description          = each.key
+  initial_group_config = var.initial_group_config
+  customer_id          = var.customer_id
+}
+
+module "optional_group" {
+  for_each = var.create_groups_holder.optional_groups != null ? tomap(var.create_groups_holder.optional_groups) : {}
```

Comment:
where are you testing that the value (the group email) is not ""?

Reply:
At 0-bootstrap/modules/groups/variables.tf on the validation blocks I'm defining that they must be following an email regex pattern.

Reply:
```hcl
  validation {
    condition     = var.create_groups_holder.create_groups == true ?
 (var.create_groups_holder.optional_groups.gcp_global_secrets_admin == "" ? true : can(regex..."
  }
 ```

`gcp_global_secrets_admin == "" ? true` it means that 

this is a valid input

```hcl
    optional_groups = {
      gcp_platform_viewer      = ""
      gcp_security_reviewer    = "gcp_security_reviewer@example.com"
      gcp_network_viewer       = ""
      gcp_scc_admin            = ""
      gcp_global_secrets_admin = "gcp_global_secrets_admin@example.com"
      gcp_audit_viewer         = ""
    }
```

will the code, the `for_each`, work in this case ? @mauro-cit

Reply:
Yes! Exactly , because if we pass these values as null, they won't be validated. Since we want to prompt to the user the exact field that failed, we must have them defined.